### DISCUSSION TITLE
Avoiding APM attribute length restrictions

### DISCUSSION THREAD

**Discussion by zilladotexe** [open-telemetry/opentelemetry-dotnet#3542] [Q&A] [answered]
Hi all,

We're trying to debug an issue where Logs aren't appearing in our APM (spans work correctly). The APM is reporting `One or more OTLP data point(s) was dropped because the length of an attribute value was over the limit.`. I can't find a way to limit attribute length in the API. I've tried setting some of the envvars mentioned in the Otel spec such as `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT` but this doesn't resolve the issue. Any ideas?

**Comment by cijothomas** [comment, marked as answer]
@alanwest has done initial work to support this! it done for Traces only now, and part of the alpha.1 release from yesterday.

**Reply by zilladotexe** [reply]
Thanks for this. Is there anything we can follow to track progress of this feature specifically for logs?

**Reply by cijothomas** [reply]
I don't think there is any spec for Logs about limiting attribute lengths.
